lib/src/main/java/org/esbtools/eventhandler/Result.java

```diff
@@ -18,10 +18,7 @@
 
 package org.esbtools.eventhandler;
 
-public interface LookupResult {
-    Object getBody();
-
-    boolean hasErrors();
-
-    DocumentEvent getAssociatedEvent();
+public interface Result<T> {
+    T get();
```

Comment:
Since this is pretty much the `java.util.concurrent.Future` interface, why not use that? Or at least name it consistently so it'll be less impact to change in the future... (pun intended :-P) (i.e. rename to `Future`).